src/crimson/common/fatal_signal.cc

```diff
@@ -68,6 +70,29 @@ static void print_segv_info(const siginfo_t* siginfo)
   std::cerr << std::flush;
 }
 
+static void print_proc_maps()
+{
+  const int fd = open("/proc/self/maps", O_RDONLY);
```

Comment:
might want to `close(fd)`? i'd suggest use `scope_guard()` and `make_scope_guard()` to do this job.

Reply:
also why not use `std::ifstream`?

Reply:
I would vote against ifstream, both for the extra complexity, and for the fact that, as we have crushed, we should as a general rule not be depended on std state.

Reply:
i'd like to understand more about the details of "complexity".

Reply:
For example: it manages a basic_streambuf, which is a pretty complex structure.

And as such, it manages 'char's, which is not exactly what we want here (no actual problem, though).
Another "purist" point: /proc/self/maps should usually be read whole in one 'read' syscall (not guaranteed to be stable from one call to the next). With ifstream - the size of the reads is not immediately obvious.

Reply:
i assume that `n` in [istream& read (char* s, streamsize n)](http://www.cplusplus.com/reference/istream/istream/read/) should do the trick?

Reply:
I doubt there is any benefit in employing higher-level constructs while I clearly see some costs: enforcing readers to go through them in detail if they want to ensure that e.g. no dynamic allocation is being made.

Reply:
> Another "purist" point: /proc/self/maps should usually be read whole in one 'read' syscall (not guaranteed to be stable from one call to the next).

Yeah, that's a valid point. Maybe we should enlarge the buffer a few times but it still won't be perfect. Even if could use dynamic memory, there is no atomic stat-alloc-read (nor stat-mmap) AFAIK.